### PR TITLE
Fix side evaluation warning

### PR DESCRIPTION
**Pull request by roigcarlo**
Fixes annoying warnings...
This is very old (picked from https://github.com/KratosMultiphysics/Kratos/tree/feature-correct-kratos-components-clang-warning) and was edited with atom, so plenty of whitespace removal... sorry for that

### CODE REVIEW

**Comment by roigcarlo** [comment]
@pooyan-dadvand Ping!

File: kratos/sources/model_part_io.cpp [hunk 1]
```diff
@@ -19,6 +19,15 @@
 
 namespace Kratos
 {
+    /// Compare component type utility
```
Comment (pooyan-dadvand) [inline kratos/sources/model_part_io.cpp:22]: This is not the equivalent to `std::is_same`?

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
There are no equivalent std trait for `isTypeIndexEq`? 

Also you should apply the class name Kratos convention.

**Comment by roigcarlo** [comment]
I think `std::is_same` is equivalent to `std::remove_cv_t<std::remove_reference_t<TypeA>> == std::remove_cv_t<std::remove_reference_t<TypeB>>`

Basically `std::is_same` does not ignore the `const` and `volatile` attributes, while comparing `typeid` does. If this is not relevant here I can change it.

Also is a function!, but the `i` is wrong right, changing

**Comment by roigcarlo** [comment]
Just for the record, we are investigating now if this code is even necessary, the PR is not halt.

**Comment by pooyan-dadvand** [comment]
I would suggest to make a test for this part and then change it to see if it is still working.

**Comment by roigcarlo** [comment]
I am closing this as the code has changed again. I will open an issue with the problem because I don't want to fix this for the third time unless we agree what to do.

**Pull request closed without merging**